es52x/src/main/scala/tech/beshu/ror/es/ReadonlyRestPlugin.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.es
+
+import java.nio.file.Path
+import java.util
+import java.util.Collections
+import java.util.function.{Supplier, UnaryOperator}
+
+import monix.execution.Scheduler
+import monix.execution.schedulers.CanBlock
+import org.elasticsearch.ElasticsearchException
+import org.elasticsearch.action.support.ActionFilter
+import org.elasticsearch.action.{ActionRequest, ActionResponse}
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry
+import org.elasticsearch.common.network.NetworkService
+import org.elasticsearch.common.settings._
+import org.elasticsearch.common.util.BigArrays
+import org.elasticsearch.common.util.concurrent.ThreadContext
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.env.Environment
+import org.elasticsearch.http.HttpServerTransport
+import org.elasticsearch.index.IndexModule
+import org.elasticsearch.index.mapper.MapperService
+import org.elasticsearch.indices.breaker.CircuitBreakerService
+import org.elasticsearch.plugins.ActionPlugin.ActionHandler
+import org.elasticsearch.plugins._
+import org.elasticsearch.rest.{RestChannel, RestHandler, RestRequest}
+import org.elasticsearch.threadpool.ThreadPool
+import org.elasticsearch.transport.Transport
+import tech.beshu.ror.Constants
+import tech.beshu.ror.configuration.RorSsl
+import tech.beshu.ror.es.dlsfls.RoleIndexSearcherWrapper
+import tech.beshu.ror.es.rradmin.rest.RestRRAdminAction
+import tech.beshu.ror.es.rradmin.{RRAdminAction, TransportRRAdminAction}
+import tech.beshu.ror.es.ssl.{SSLNetty4HttpServerTransport, SSLNetty4InternodeServerTransport}
+import tech.beshu.ror.es.utils.ThreadRepo
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+@Inject
+class ReadonlyRestPlugin(s: Settings, p: Path)
```

Comment:
It may not work. I see failed builds (but I didn't check reasons of the fail), but maybe this is the same what I experienced some time ago. There is a problem with `@Inject` annotation and scala classes constructor. I've found workaround of the issue - you have to create auxiliary constructor and annotate it. See, eg: https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/blob/3006a081fe18c3978a8f4af21235b32075d33540/es70x/src/main/scala/tech/beshu/ror/es/rradmin/TransportRRAdminAction.scala#L38